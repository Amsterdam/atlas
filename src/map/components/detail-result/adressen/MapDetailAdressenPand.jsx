import React from 'react';
import PropTypes from 'prop-types';

import MapDetailResultItem from '../MapDetailResultItem';
import MapDetailResultWrapper from '../MapDetailResultWrapper';

const MapDetailAdressenPand = ({ panoUrl, pand }) => (
  <MapDetailResultWrapper
    panoUrl={panoUrl}
    subTitle={pand.label}
    title="Pand"
  >
    <ul className="map-detail-result__list">
      <MapDetailResultItem
        label="Oorspronkelijk bouwjaar"
        value={pand.year || 'onbekend'}
      />
      <MapDetailResultItem
        label="Status"
        value={pand.status}
      />
    </ul>
  </MapDetailResultWrapper>
);

MapDetailAdressenPand.propTypes = {
  pand: PropTypes.shape({
    label: PropTypes.string,
    status: PropTypes.string,
    year: PropTypes.string
  }).isRequired,
  panoUrl: PropTypes.string.isRequired
};

export default MapDetailAdressenPand;
