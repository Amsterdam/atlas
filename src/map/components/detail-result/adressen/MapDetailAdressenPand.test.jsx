import React from 'react';
import { shallow } from 'enzyme';

import MapDetailAdressenPand from './MapDetailAdressenPand';

describe('MapDetailAdressenPand', () => {
  it('should render everything', () => {
    const pand = {
      label: 'Pand label',
      status: {
        ommschrijving: 'Pand status',
        code: '31'
      },
      year: '2020'
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailAdressenPand
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        pand={pand}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render unknown for empty year', () => {
    const pand = {
      label: 'Pand label',
      status: 'Pand status',
      year: ''
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailAdressenPand
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        pand={pand}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render unknown without year', () => {
    const pand = {
      label: 'Pand label',
      status: 'Pand status'
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailAdressenPand
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        pand={pand}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
