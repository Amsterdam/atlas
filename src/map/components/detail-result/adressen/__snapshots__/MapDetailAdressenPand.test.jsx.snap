// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapDetailAdressenPand should render everything 1`] = `
<Connect(MapDetailResultWrapper)
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Pand label"
  title="Pand"
>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailResultItem
      hasMultiline={false}
      label="Oorspronkelijk bouwjaar"
      value="2020"
    />
    <MapDetailResultItem
      hasMultiline={false}
      label="Naam"
      value="Gebouw!"
    />
    <MapDetailResultStatusItem
      label="Status"
      status=""
      value=""
    />
  </ul>
</Connect(MapDetailResultWrapper)>
`;

exports[`MapDetailAdressenPand should render unknown for empty year and empty name 1`] = `
<Connect(MapDetailResultWrapper)
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Pand label"
  title="Pand"
>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailResultItem
      hasMultiline={false}
      label="Oorspronkelijk bouwjaar"
      value="onbekend"
    />
    <MapDetailResultItem
      hasMultiline={false}
      label="Naam"
      value=""
    />
    <MapDetailResultStatusItem
      label="Status"
      status=""
      value=""
    />
  </ul>
</Connect(MapDetailResultWrapper)>
`;

exports[`MapDetailAdressenPand should render unknown without year and name 1`] = `
<Connect(MapDetailResultWrapper)
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Pand label"
  title="Pand"
>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailResultItem
      hasMultiline={false}
      label="Oorspronkelijk bouwjaar"
      value="onbekend"
    />
    <MapDetailResultItem
      hasMultiline={false}
      label="Naam"
      value=""
    />
    <MapDetailResultStatusItem
      label="Status"
      status=""
      value=""
    />
  </ul>
</Connect(MapDetailResultWrapper)>
`;
